packages/gatsby-source-drupal/src/gatsby-node.js

```diff
@@ -268,143 +278,59 @@ exports.sourceNodes = async (
-        searchParams: params,
-        responseType: `json`,
-      },
-    ])
-    allData = await Promise.all(
-      _.map(res.body.links, async (url, type) => {
-        const dataArray = []
-        if (disallowedLinkTypes.includes(type)) return
-        if (!url) return
-        if (!type) return
-
-        // Lookup this type in our list of language alterable entities.
-        const isTranslatable = languageConfig.translatableEntities.some(
-          entityType => entityType === type
-        )
-
-        const getNext = async url => {
-          if (typeof url === `object`) {
-            // url can be string or object containing href field
-            url = url.href
-
-            // Apply any filters configured in gatsby-config.js. Filters
-            // can be any valid JSON API filter query string.
-            // See https://www.drupal.org/docs/8/modules/jsonapi/filtering
-            if (typeof filters === `object`) {
-              if (filters.hasOwnProperty(type)) {
-                url = new URL(url)
-                const filterParams = new URLSearchParams(filters[type])
-                const filterKeys = Array.from(filterParams.keys())
-                filterKeys.forEach(filterKey => {
-                  // Only add filter params to url if it has not already been
-                  // added.
-                  if (!url.searchParams.has(filterKey)) {
-                    url.searchParams.set(filterKey, filterParams.get(filterKey))
-                  }
-                })
-                url = url.toString()
-              }
-            }
-          }
-
-          let d
-          try {
-            d = await requestQueue.push([
-              url,
-              {
-                username: basicAuth.username,
-                password: basicAuth.password,
-                headers,
-                responseType: `json`,
-              },
-            ])
-          } catch (error) {
-            if (error.response && error.response.statusCode == 405) {
-              // The endpoint doesn't support the GET method, so just skip it.
-              return
-            } else {
-              console.error(`Failed to fetch ${url}`, error.message)
-              console.log(error)
-              throw error
-            }
-          }
-          dataArray.push(...d.body.data)
-          // Add support for includes. Includes allow entity data to be expanded
-          // based on relationships. The expanded data is exposed as `included`
-          // in the JSON API response.
-          // See https://www.drupal.org/docs/8/modules/jsonapi/includes
-          if (d.body.included) {
-            dataArray.push(...d.body.included)
-          }
-          if (d.body.links && d.body.links.next) {
-            await getNext(d.body.links.next)
+  const listResponse = await requestQueue.push([
+    urlJoin(baseUrl, 'gatsby/content-list'),
+    {
+      headers,
+    }
+  ])
+  const listResponseBody = JSON.parse(listResponse.body)
```

Comment:
For whatever reason, passing `responseType: 'json'` to got and letting it do the JSON.parse results in some very weird artifacts in some environments. It worked 100% of the time on my local builds but doing a build in CI failed.

Reply:
it's almost certainly that somehow your CI version isn't getting the right version of `got`. Probably because you're running this in a local plugin which means that it's whatever version of `got` is resolving in your site which because Gatsby's dependency tree unfortunately contains older versions of Got, isn't going to be predictable atm. Probably best for testing to publish the module to a temp npm package so that it can resolve its own dependencies correctly.